src/vllm_tgis_adapter/grpc/grpc_server.py

```diff
@@ -257,9 +257,22 @@ async def Generate(
             )
 
         # TODO handle cancellation
-        result_generator: AsyncIterator[tuple[int, RequestOutput]] = (
-            merge_async_iterators(*generators)
-        )
+        result_generator: AsyncIterator[tuple[int, RequestOutput]]
+        try:
+            result_generator = merge_async_iterators(*generators)
+        except TypeError as exc:  # vllm > 0.5.4
+            if (
+                "merge_async_iterators() missing 1 required keyword-only argument: "
+                "'is_cancelled'" not in str(exc)
+            ):
+                raise
+
+            async def is_cancelled() -> bool:
+                return context.cancelled()
+
+            result_generator = merge_async_iterators(
+                *generators, is_cancelled=is_cancelled
+            )
```

Comment:
This does not really do much at this point, since we never call [`context.cancel()`](https://grpc.github.io/grpc/python/grpc_asyncio.html#grpc.aio.RpcContext.cancel) , so `context.cancelled()` always return `False` ([docs](https://grpc.github.io/grpc/python/grpc_asyncio.html#grpc.aio.RpcContext.cancelled))